### PR TITLE
[flutter_tools] ensure cache directory is created

### LINKED ISSUE #84960: [flutter_tools] locally built tools expect `cache` folder with Windows
Steps:
1. Clone `flutter` project with Windows 10
2. Run `bin\flutter_tools.dart` from Android Studio

Expected: normal output
Actual:

```
Failed to open or create the artifact cache lockfile: "FileSystemException: Cannot open file, path = 'C:\use\flutter\clone\flutter\bin\cache\lockfile' (OS Error: The system cannot find the path specified.
, errno = 3)"
Please ensure you have permissions to create or open C:\use\flutter\clone\flutter\bin\cache\lockfile
Failed to open or create the lockfile

Process finished with exit code 1

```

The reason is because `cache` folder doesn't exist, if it is manually created, it would be fine.

### PR DESCRIPTION
**Pull request by asashour**
Fixes #84960

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/lib/src/cache.dart [hunk 1]
```diff
@@ -302,8 +302,15 @@ class Cache {
       return;
     }
     assert(_lock == null);
-    final File lockFile =
-      _fileSystem.file(_fileSystem.path.join(flutterRoot!, 'bin', 'cache', 'lockfile'));
+    final Directory cacheDir =
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/cache.dart:305]: The cache dir is usually created by the entrypoint script in bin/flutter or bin/dart. What is the usecase for running the tool with a partially set up SDK directly?

Comment (hamad22h) [reply packages/flutter_tools/lib/src/cache.dart:305]: F

Comment (asashour) [reply packages/flutter_tools/lib/src/cache.dart:305]: > The cache dir is usually created by the entrypoint script in bin/flutter or bin/dart. What is the usecase for running the tool with a partially set up SDK directly?

The use case is running the flutter tools from a cloned source directly, e.g. I was investigating and trying to debug the `flutter` CLI and came across this.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/cache.dart:305]: Were you running it with a different `dart` than what comes with it? That is the only way you could get here without the cache dir existing

Comment (asashour) [reply packages/flutter_tools/lib/src/cache.dart:305]: I see your point.

I cloned `flutter` project again, and used the `dart-sdk` from the originally installed `master`.

Please close this issue if the user must always run `flutter.bat` whenever a clone is done.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/cache.dart:305]: This is working as intended, the bootstrap scripts are not optional.

**Pull request closed without merging**